src/public/class-wordlift-term-jsonld-adapter.php

```diff
@@ -52,38 +53,90 @@ public function __construct( $entity_uri_service, $jsonld_service ) {
 	}
 
 	/**
-	 * Hook to `wp_head` to print the JSON-LD.
+	 * Adds carousel json ld data to term page if the conditions match
 	 *
-	 * @since 3.20.0
+	 * @param $jsonld array JsonLd Array.
+	 *
+	 * @return array
 	 */
-	public function wp_head() {
+	public function get_carousel_jsonld( $jsonld ) {
+		global $wp_query;
 
-		// Bail out if it's not a category page.
-		if ( ! is_tax() && ! is_category() ) {
-			return;
+		if ( ! is_array( $wp_query->posts ) ) {
+			// Bail out if no posts are present.
+			return $jsonld;
 		}
 
-		$term    = get_queried_object();
-		$term_id = $term->term_id;
+		$posts = array_map( function ( $post ) {
+			return $post->ID;
+		}, $wp_query->posts );
 
-		// The `_wl_entity_id` are URIs.
-		$entity_ids         = get_term_meta( $term_id, '_wl_entity_id' );
-		$entity_uri_service = $this->entity_uri_service;
+		if ( count( $posts ) < 2 ) {
+			return $jsonld;
+		}
+		// More than 2 items are present, so construct the jsonld data
+		$jsonld['@context']        = 'https://schema.org';
+		$jsonld['@type']           = 'ItemList';
+		$jsonld['itemListElement'] = array();
+		$position                  = 1;
+		foreach ( $posts as $post_id ) {
+			$post_jsonld = $this->jsonld_service->get_jsonld( false, $post_id );
```

Comment:
@naveen17797 I think `$this->jsonld_service->get_jsonld( ... )` returns an array, in which case the first item would be the post JSON-LD and subsequent items would be entities referenced by that post. The first item should be added to `item`, the referenced items should be added to the root of the JSON-LD structure.